app/services/course/assessment/submission/auto_grading_service.rb

```diff
@@ -38,6 +50,13 @@ def grade_answers(submission)
   def grade_answer(answer)
     raise ArgumentError if answer.changed?
     answer.auto_grade!
+    # Catch errors if answer is in attempting state, caused by a race condition where
+    # a new attempting answer is created while the submission is finalised, but before the
+    # autograding job is executed.
+  rescue IllegalStateError
```

Comment:
The intent is to reduce the number of special scenarios. If every `auto_grade!` on an `attempting` answer is going to be specially rescued and forced to `submitted`, then it might as well be valid to `auto_grade!` an `attempting` answer.

Is there anywhere that needs `auto_grade!` to fail on an `attempting` answer?